inspirehep/modules/orcid/cache.py

```diff
@@ -0,0 +1,153 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2018 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+from __future__ import absolute_import, division, print_function
+
+import flask
+import hashlib
+
+from flask import current_app as app
+from redis import StrictRedis
+from StringIO import StringIO
+
+from .converter import OrcidConverter
+from .exceptions import EmptyPutcodeError
+
+
+class OrcidCache(object):
+    def __init__(self, orcid, recid):
+        """
+        Orcid cached data.
+
+        Args:
+            orcid (string): orcid identifier.
+        """
+        self.orcid = orcid
+        self.recid = recid
+        self._cached_hash_value = None
+        self._new_hash_value = None
+
+    @property
+    def redis(self):
+        redis = getattr(flask.g, 'redis_client', None)
+        if redis is None:
+            url = app.config.get('CACHE_REDIS_URL')
+            redis = StrictRedis.from_url(url)
+            flask.g.redis_client = redis
+        return redis
+
+    @property
+    def _key(self):
+        """Return the string 'orcidcache:``orcid_value``:``recid``'"""
+        return 'orcidcache:{}:{}'.format(self.orcid, self.recid)
+
+    def write_work_putcode(self, putcode, inspire_record=None):
+        """
+        Write the putcode and the hash for the given (orcid, recid).
+
+        Args:
+            putcode (string): the putcode used to push the record to ORCID.
+            inspire_record (InspireRecord): InspireRecord instance. If provided,
+             the hash for the record content is re-computed.
+        """
+        if not putcode:
+            raise EmptyPutcodeError
```

Comment:
why not simply `ValueError('putcode must not be empty')`?

Reply:
Valid point, I will do that